### DISCUSSION TITLE
$output argument in post-controller events

### DISCUSSION THREAD

**Discussion by vkulish04** [opencart/opencart#11500] [Q&A]
Something that I don't completely understand.
https://github.com/opencart/opencart/blob/ebe4f81f2a58e01ee9e1000478444b62d04be14b/upload/system/framework.php#L248
here variable named $output is used as an argument to post-controller events. Where it is initialized?

**Comment by mhcwebdesign** [comment]
See file system/engine/loader.php at lines 75-76:

```
		$action = new \Opencart\System\Engine\Action($route);
		$output = $action->execute($this->registry, $args);

```

**Reply by vkulish04** [reply]
Oh, got it, thanks

**Reply by vkulish04** [reply]
But the `$output` variable in that case is local, and is not initialized anywhere.
And it is found only in `$event->trigger('controller/' . $trigger . '/after', [&$route, &$args, &$output]);`.
Why is the bath not initialized?

**Comment by mhcwebdesign** [comment]
You are right, see this issue: https://github.com/opencart/opencart/issues/11542

The controller method called by the dispatcher in the system/framework.php via an $action->execute(...) don't return an output string, they are usually of return type void, unlike the ones called from the system/engine/loader.php.
